packages/gguf/src/gguf.ts

```diff
@@ -293,5 +295,41 @@ export async function gguf(
 		});
 	}
 
-	return { metadata, tensorInfos };
+	let paramCount: number | undefined = undefined;
+
+	if (params?.calculateParamCount) {
+		paramCount = tensorInfos
+			.map(({ shape }) => shape.reduce((acc, val) => acc * Number(val), 1))
+			.reduce((acc, val) => acc + val, 0);
+
+		const ggufShardFileInfo = parseGgufShardFile(url);
+		if (ggufShardFileInfo) {
```

Comment:
```suggestion
		if (ggufShardFileInfo && params?.fetchAllShards) {
```

IMO this behavior (fetching other shards magically) is quite unintuitive, so I would at least add a special flag for it.

Or maybe better API-wise to split this into another method